### PR TITLE
Fix BodyTopLevelPropertiesOnly to support get-only APIs

### PR DESCRIPTION
**Pull request by bdefoy**
BodyTopLevelProperties currently does not catch invalid top-level properties for get only APIs. This PR adds behavior to the rule to examine get operations in addition to the put and patch operations that it already examined.

### CODE REVIEW

File: packages/rulesets/generated/spectral/az-dataplane.js
Comment (bdefoy) [inline packages/rulesets/generated/spectral/az-dataplane.js:1]: Note for reviewers: these changes were generated when I ran the build.

**Review by rkmanda — APPROVED** [review]
:shipit:

File: packages/rulesets/generated/spectral/az-dataplane.js
Comment (mikeharder) [reply packages/rulesets/generated/spectral/az-dataplane.js:1]: I think this is fallout from https://github.com/Azure/azure-openapi-validator/pull/703.

@bdefoy: Please revert the changes to az-dataplane.js manually in your PR.

@konrad-jamrozik: We will need a better solution for the long term.

**Review by mikeharder — CHANGES_REQUESTED** [review]
Revert changes to az-dataplane.js

**Review by mikeharder — CHANGES_REQUESTED** [review]
I think you also want to bump the package version here, so you can publish a new version after merging:

https://github.com/azure/azure-openapi-validator/blob/main/packages/rulesets/package.json#L3

**Review by konrad-jamrozik — APPROVED** [review]

**Comment by konrad-jamrozik** [comment]
@bdefoy note the current deployment process is here:

https://github.com/Azure/azure-openapi-validator/pull/669/files#r1515048617

**Review by mikeharder — APPROVED** [review]

**Pull request merged**